Documentation/benchmarks/etcd-storage-memory-benchmark.md

```diff
@@ -0,0 +1,95 @@
+# Storage Memory Usage Benchmark
+
+The storage(todo: link to design doc about storage) has two components that consume physical memory: the _in-memory index_, and the _page cache_ for on-disk data.
+
+The in-memory index holds all the keys in a [B-tree][btree] data structure, along with pointers to the on-disk data (the values). Each key in the B-tree may contain multiple pointers, pointing to different versions of its values. The theoretical memory consumption of the in-memory index can hence be approximated with the formula:
+
+`N * (c1 + avg_key_size) + N * (avg_versions_of_key) * (c2 + size_of_pointer)`
```

Comment:
`along with pointers to the on-disk data`. This pointer is not what you were referring to.